tsdb/chunks/chunks.go

```diff
@@ -217,22 +217,33 @@ func (w *Writer) cut() error {
 	return nil
 }
 
-func cutSegmentFile(dirFile *os.File, magicNumber uint32, chunksFormat byte, allocSize int64) (headerSize int, newFile *os.File, seq int, err error) {
+func cutSegmentFile(dirFile *os.File, magicNumber uint32, chunksFormat byte, allocSize int64) (headerSize int, newFile *os.File, seq int, returnErr error) {
 	p, seq, err := nextSequenceFile(dirFile.Name())
 	if err != nil {
-		return 0, nil, 0, err
+		return 0, nil, 0, errors.Wrap(err, "next sequence file")
 	}
-	f, err := os.OpenFile(p, os.O_WRONLY|os.O_CREATE, 0666)
+	ptmp := p + ".tmp"
+	f, err := os.OpenFile(ptmp, os.O_WRONLY|os.O_CREATE, 0666)
 	if err != nil {
-		return 0, nil, 0, err
-	}
+		return 0, nil, 0, errors.Wrap(err, "open temp file")
+	}
+	defer func() {
+		if returnErr != nil {
+			var merr tsdb_errors.MultiError
+			merr.Add(returnErr)
+			merr.Add(f.Close())
```

Comment:
Sorry to beat this seemingly dead horse. But `f` can be `nil` if opening the final file in L266 fails. You are reusing `f` there, which is probably OK as you actually want to close the new `f` in case there is an error in L271.

I think you have to add a check here for `f != nil`. And then you could set `f = nil` in L261 so that you don't try to close the closed file in case the Rename in 262 fails.

Reply:
Done! Thanks